src/app/core/duplicate-records/duplicate-records.service.spec.ts

```diff
@@ -0,0 +1,70 @@
+import { TestBed } from '@angular/core/testing';
+import { DuplicateRecordsService } from './duplicate-records.service';
+import { EntityMapperService } from '../entity/entity-mapper/entity-mapper.service';
+import { DatabaseEntity, EntityRegistry } from '../entity/database-entity.decorator';
+import { EntitySchemaService } from '../entity/schema/entity-schema.service';
+import { Database } from '../database/database'; 
+import { SessionService } from '../session/session-service/session.service';
+import { Entity } from '../entity/model/entity';
+import { DatabaseField } from '../entity/database-field.decorator';
+
+describe('DuplicateRecordsService', () => {
+  let service: DuplicateRecordsService;
+  let entityMapperService: EntityMapperService;
+  let entityTypes: EntityRegistry;
+  let entitySchemaService: EntitySchemaService;
+  let database: Database; 
+  let sessionService: SessionService;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+        DuplicateRecordsService,
+        EntityMapperService,
+        EntityRegistry,
```

Comment:
```suggestion
  @DatabaseEntity("your_schema_name")
  class YourTestClass extends Entity {}

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [CoreModule],
      providers: [
        DuplicateRecordsService,
        EntityMapperService,
        { provide: EntityRegistry, useValue: entityRegistry },
        ComponentRegistry,
```

This will make the test work:
- the automatic registration for any `@DatabaseEntity()` annotation happens in the global variable `entityRegistry`, so this has to be provided specifically ...
- to have the data types for DatabaseFields, the CoreModule registers a lot of the common ones. And that in turn requires ComponentRegistry also ...

An alternative would be to simply import our `MockedTestingModule` here, which brings all of those related things with it. But it also imports a lot of other things, making it more of an integration than a unit test then.